### PR TITLE
DEF_HELPER_N -> DEF_HELPER_FLAGS_N

### PR DESCRIPTION
**Pull request by mferst**
In our "PowerISA Vector/VSX instruction batch" patch series, rth noted[1] that helpers that only access vector registers should be declared with `DEF_HELPER_FLAGS_N` and `TCG_CALL_NO_RWG`. We fixed helpers in that series, but there are older helpers that could use the same optimization.

Guided by the presence of `env` as the first argument, in patches 1~4 we change helpers that do not have access to the `cpu_env` pointer to modify any globals. Then, we change other helpers that receive `cpu_env` but do not use it and apply the same fix, taking the opportunity to move them to decodetree.

[1] https://lists.gnu.org/archive/html/qemu-ppc/2022-02/msg00568.html

### CODE REVIEW

**Review by vcoracolombo — CHANGES_REQUESTED** [review]
Looks really good, nice job. Just one nitpick:  target/ppc: Use TCG_CALL_NO_RWG_SE in fsel helper  commit message doesn't say anything about moving to decodetree. I think we could add this :)

Otherwise, just fix the macro call order

File: target/ppc/fpu_helper.c [hunk 1]
```diff
@@ -916,8 +916,7 @@ float64 helper_frsqrtes(CPUPPCState *env, float64 arg)
 }
 
 /* fsel - fsel. */
-uint64_t helper_fsel(CPUPPCState *env, uint64_t arg1, uint64_t arg2,
-                     uint64_t arg3)
+uint64_t helper_FSEL(uint64_t arg1, uint64_t arg2, uint64_t arg3)
```
Comment (luporl) [inline target/ppc/fpu_helper.c:919]: Why it's necessary to change part of the helpers' name to uppercase?

**Review by luporl — APPROVED** [review]
I haven't checked all the details, but the changes look good to me.

File: target/ppc/fpu_helper.c (hunk 1 above)
Comment (luporl) [reply target/ppc/fpu_helper.c:919]: Ok, thanks for the clarification.

**Review by alqotel — APPROVED** [review]
LGTM!

**Comment by mferst** [comment]
Merged: https://lists.gnu.org/archive/html/qemu-ppc/2022-05/msg00453.html

**Pull request closed without merging**